posthog/models/feature_flag.py

```diff
@@ -359,22 +365,25 @@ def get_match(self, feature_flag: FeatureFlag) -> FeatureFlagMatch:
             match=False, reason=highest_priority_evaluation_reason, condition_index=highest_priority_index
         )
 
-    def get_matches(self) -> Tuple[Dict[str, Union[str, bool]], Dict[str, dict]]:
+    def get_matches(self) -> Tuple[Dict[str, Union[str, bool]], Dict[str, dict], Dict[str, Optional[Dict]]]:
         flags_enabled = {}
         flag_evaluation_reasons = {}
+        flag_payloads = {}
         for feature_flag in self.feature_flags:
             try:
                 flag_match = self.get_match(feature_flag)
                 if flag_match.match:
                     flags_enabled[feature_flag.key] = flag_match.variant or True
+                    if flag_match.payload:
+                        flag_payloads[feature_flag.key] = flag_match.payload
```

Comment:
Now decide v3 would return an extra field `featureFlagPayloads` that looks like 

```
{
...
"featureFlagPayloads": {
     {flag-key}: payload
}
...
}
```

This way flags will exist on all previous versions of decide but newer versions will be able to retrieve a payload associated with the match

Reply:
Makes sense for regular flags, but multivariant flags might require a payload per variant?

Reply:
I'm currently thinking no because if you call `posthog.getFeatureFlagPayload(flag-key)`, multivariant or not it will just return nothing or the payload from the matched variant. But I'm not opposed to making it super explicit where you would specify the key and the variant key to get the payload

Reply:
Aha, fair, makes sense 😅

Reply:
Yeah, this is good, reduces dumb client-side mistakes by quite a bit, makes it easy to use. 💯